hs_core/models.py

```diff
@@ -2252,16 +2277,71 @@ def has_required_content_files(self):
         else:
             return True
 
+    @property
+    def readme_file(self):
+        """Returns a resource file that is at the root with a file name of either
+        'readme.txt' or 'readme.md' (filename is case insensitive). If no such file then None
+        is returned. If both files exist then resource file for readme.md is returned"""
+
+        res_files_at_root = [f for f in self.files.all() if f.file_folder is None]
+        readme_txt_file = None
+        readme_md_file = None
+        for res_file in res_files_at_root:
+            if res_file.file_name.lower() == 'readme.md':
+                readme_md_file = res_file
+            elif res_file.file_name.lower() == 'readme.txt':
+                readme_txt_file = res_file
+            if readme_md_file is not None:
+                break
+
+        if readme_md_file is not None:
+            return readme_md_file
+        else:
+            return readme_txt_file
+
+    @property
+    def has_readme_file(self):
+        """Returns True if the resource has a readme.md or readme.txt file at the root, otherwise
+        False."""
+
+        return self.readme_file is not None
+
+    def get_readme_file_content(self):
+        """Gets the content of the readme file. If both a readme.md and a readme.txt file exist,
+        then the content of the readme.md file is returned.
+        :raises If there is no readme file, ObjectDoesNotExist is raised
+        """
+        if self.readme_file is None:
+            raise ObjectDoesNotExist
+        else:
+            if self.readme_file.extension.lower() == '.md':
+                return {'content': markdown(self.readme_file.read().decode('utf-8')),
+                        'file_name': self.readme_file.file_name}
+            else:
+                return {'content': self.readme_file.read(), 'file_name': self.readme_file.file_name}
```

Comment:
Why do we need to support txt files as well?

Reply:
@horsburgh probably knows the answer.

Reply:
@sblack-usu and @pkdash - when we originally discussed readme functionality, we started with readme.txt, but then people wanted to do rich text formatting, so we added the ability to do markdown via readme.md. I think it's still valid for someone to have a readme.txt file that does not contain markdown formatting. So, the rule is, a readme.md or a readme.txt file can be uploaded, and either one will be displayed on the page if only one file is uploaded. If neither are uploaded, nothing is displayed. If both are uploaded, the readme.md gets preference over a readme.txt file for display.

Reply:
My only argument for removing txt file support is to keep things simpler both in the logic and usability.  It's not much more logic to support txt files, so not a huge deal, but I'm not sure we really gain much by supporting both and then we also have to clarify to users when one file is displayed over the other.  I'm not going to hold anything up over this, just voicing my opinion that I think it's sufficient to only support markdown.

Reply:
@sblack-usu - For simplicity sake, I would be OK with letting the rendering of readme.txt go as long as we have buy-in from the team. There's sort of a history associated with readme.txt as a place to put documentation of scientific datasets. I'm not sure a lot of users will know what a ".md" file is (although it is becoming more common as more people are using GitHub). Users could still upload a readme.txt file, it just wouldn't be rendered on the page. If we don't support both, we likely need some indication somewhere that says - "Upload a readme.md file with documentation for your resource. The text will be displayed on the resource landing page and formatting is supported via Markdown." 

Or something like that...